src/psyclone/transformations.py

```diff
@@ -412,99 +463,203 @@ def name(self):
                                      an inter-grid kernel.
-        '''
-        LoopFuseTrans._validate(self, node1, node2)
-
-        from psyclone.dynamo0p3 import VALID_FUNCTION_SPACES
-        try:
-
-            # Check that we don't have an inter-grid kernel
-            check_intergrid(node1)
-            check_intergrid(node2)
-
-            if node1.field_space.orig_name in VALID_FUNCTION_SPACES and \
-               node2.field_space.orig_name in VALID_FUNCTION_SPACES:
-                if node1.field_space.orig_name != node2.field_space.orig_name:
-                    if same_space:
-                        info = (
-                            " Note, the same_space flag was set, but "
-                            "does not apply because neither field is "
-                            "ANY_SPACE.")
-                    else:
-                        info = ""
-                    raise TransformationError(
-                        "Error in DynamoLoopFuse transformation. "
-                        "Cannot fuse loops that are over different spaces: "
-                        "{0} {1}.{2}".format(node1.field_space.orig_name,
-                                             node2.field_space.orig_name,
-                                             info))
-            else:  # one or more of the function spaces is any_space
-                if not same_space:
-                    raise TransformationError(
-                        "DynamoLoopFuseTrans. One or more of the iteration "
-                        "spaces is unknown ('any_space') so loop fusion might "
-                        "be invalid. If you know the spaces are the same then "
-                        "please set the 'same_space' optional argument to "
-                        "True.")
-            if node1.upper_bound_name != node2.upper_bound_name:
-                raise TransformationError(
-                    "Error in DynamoLoopFuse transformation. The upper bound "
-                    "names are not the same. Found '{0}' and '{1}'".
-                    format(node1.upper_bound_name, node2.upper_bound_name))
-            if node1.upper_bound_halo_depth != node2.upper_bound_halo_depth:
+        :raises TransformationError: if one or both function spaces have \
+                                     invalid names.
+        :raises TransformationError: if the `same_space` flag was set, but \
+                                     does not apply because neither field \
+                                     is on `ANY_SPACE` or the spaces are not \
+                                     the same.
+        :raises TransformationError: if one or more of the iteration spaces
+                                     is unknown (`ANY_SPACE`) and the \
+                                     `same_space` flag is not set to `True`.
+        :raises TransformationError: if the loops are over different spaces \
```

Comment:
You explained that this test is only when iterating over cells. Please explain this in the description so people know the context of the test e.g. add "and the loops both iterate over cells." at the end.

Reply:
Thanks for the suggestion. Added.

Reply:
Great. There was a missing ```\``` which I've just added as it is just a very minor change.